packages/headless/src/api/knowledge/createAnswerAPISlice.ts

```diff
@@ -0,0 +1,126 @@
+import {
+  EventSourceMessage,
+  fetchEventSource,
+} from '@microsoft/fetch-event-source';
+import {ConfigurationSection} from '../../state/state-sections';
+import {createAbortController} from '../../utils/abort-controller-polyfill';
+import {baseKnowledgeAPISlice} from './baseKnowledgeAPISlice';
+
+type StateNeededByKnowledgeAPI = ConfigurationSection;
+
+interface ErrorPayload {
+  message?: string;
+  code?: number;
+}
+
+class FatalError extends Error {
+  constructor(public payload: ErrorPayload) {
+    super(payload.message);
+  }
+}
+
+const abortController = createAbortController();
+let retryCount = 0;
+
+const data = {
+  answerStyle: '',
```

Comment:
Most (all?) of these properties can be strongly typed, and are already defined in the project in `api/generated-answer`.

We should refactor and reuse as much as possible.